src/plugins/gmx/gmxplugin.cpp

```diff
@@ -0,0 +1,154 @@
+/*
+ * GMX Tiled Plugin
+ * Copyright 2014, Thorbjørn Lindeijer <thorbjorn@lindeijer.nl>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "gmxplugin.h"
+
+#include "map.h"
+#include "tile.h"
+#include "tilelayer.h"
+#include "mapobject.h"
+#include "objectgroup.h"
+
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QSaveFile>
+
+using namespace Tiled;
+using namespace Gmx;
+
+GmxPlugin::GmxPlugin()
+{
+}
+
+bool GmxPlugin::write(const Map *map, const QString &fileName)
+{
+    QSaveFile file(fileName);
+
+    if (!file.open(QIODevice::WriteOnly | QIODevice::Text)) {
+        mError = tr("Could not open file for writing.");
+        return false;
+    }
+
+    QString instanceData;
+    QString tileData;
+
+    file.write("<!--This Document is generated by GameMaker, if you edit it by hand then you do so at your own risk!-->\n");
+    file.write("<room>\n");
+    file.write("  <caption></caption>\n");
+    file.write("  <width>" + QByteArray::number(map->tileWidth() * map->width()) + "</width>\n");
+    file.write("  <height>" + QByteArray::number(map->tileHeight() * map->height()) + "</height>\n");
+    file.write("  <vsnap>" + QByteArray::number(map->tileWidth()) + "</vsnap>\n");
+    file.write("  <hsnap>" + QByteArray::number(map->tileHeight()) + "</hsnap>\n");
+    file.write("  <isometric>" + QByteArray::number((map->orientation() == Tiled::Map::Orientation::Isometric ? 1 : 0)) + "</isometric>\n");
```

Comment:
Since you're writing an XML file, please use [QXmlStreamWriter](http://doc.qt.io/qt-5/qxmlstreamwriter.html) instead. It will automatically take care of the formatting, the encoding and any necessary escaping of values.